### PR TITLE
Extend `tsh clusters` output with "Cluster Type" and "Selected" columns

### LINKED ISSUE #4965: Add root/leaf distinction to `tsh clusters`
### Feature Request
`tsh clusters` doesn't give much useful information at the moment compared to the web UI (which tells you root vs leaf). It also doesn't immediately tell you at a glance what cluster you're logged into.

```
~ » tsh clusters
Cluster Name                  Status 
----------------------------- ------ 
ip-172-31-52-229-ec2-internal online 
ip-172-31-49-164-ec2-internal online 
```

Maybe we could improve it to something like this?

```
~ » tsh clusters
Cluster Name                  Status Cluster Type Active
----------------------------- ------ ------------ ------
ip-172-31-52-229-ec2-internal online root         
ip-172-31-49-164-ec2-internal online leaf         *
```

Could be a `-v` flag or just more information by default.

### Motivation
Easier to use `tsh` from the command line.

### Who's it for?
OSS User, Pro, Enterprise

### PR DESCRIPTION
**Pull request by andrejtokarcik**
Fixes #4965, implementing the output format proposed therein. The additional columns are printed by default, it didn't seem necessary to introduce a `-v` flag.

### CODE REVIEW

File: tool/tsh/tsh.go [hunk 1]
```diff
@@ -1034,31 +1034,45 @@ func onListClusters(cf *CLIConf) {
...
+		var rootErr, leafErr error
+		rootCluster, rootErr = proxyClient.GetRootCluster()
+		leafClusters, leafErr = proxyClient.GetLeafClusters(cf.Context)
+		return trace.NewAggregate(rootErr, leafErr)
 	})
 	if err != nil {
 		utils.FatalError(err)
 	}
+
+	profile, _, err := client.Status("", cf.Proxy)
```
Comment (r0mant) [inline tool/tsh/tsh.go:1055]: `profile` can be returned as nil here even if err is nil so you prob. need to handle this scenario as well.

File: tool/tsh/tsh.go [hunk 2]
```diff
@@ -1034,31 +1034,45 @@ func onListClusters(cf *CLIConf) {
...
+	if err != nil {
+		utils.FatalError(err)
+	}
+	showActive := func(cluster services.RemoteCluster) string {
```
Comment (r0mant) [inline tool/tsh/tsh.go:1059]: I would check with @benarent how we want to display things that are active in the CLI so we're consistent. I think for profiles for example, the active one is marked with a `> ` symbol in front of it. I'm doing similar thing with database access.

**Review by r0mant — COMMENTED** [review]
Curious what others think, but I wonder why you opted for a more complicated approach with splitting the proxy GetAllClusters into 2 methods, rather than just keeping the API as-is and comparing each cluster name with the name of the root cluster.

Something like:

```go
rootClusterName, err := proxyClient.RootClusterName()
// handle err

sort.Slice(clusters, func(i, j int) bool { clusters[i].GetName() == rootClusterName })
for _, cluster := range clusters {
  kind := "root"
  if cluster.GetName() != rootClusterName {
    kind = "leaf"
  }
  t.AddRow(...)
}
```

Think the implementation would be a bit simpler/cleaner and proxy client interface wouldn't need to change.

**Review by awly — COMMENTED** [review]
I prefer the `GetLeafClusters` over `GetAllClusters`, so that we don't have to pretend like the local (root) cluster is a `services.RemoteCluster`.
But maybe remove `GetRootCluster`, and call `proxyClient.RootClusterName` directly in tsh.

Wrapping root cluster in a `services.RemoteCluster` feels like we're fighting the type system for the sake of some code de-duplication in tsh.

**Comment by andrejtokarcik** [comment]
> Curious what others think, but I wonder why you opted for a more complicated approach with splitting the proxy GetAllClusters into 2 methods.

My thought process was basically the following:

1. The notions of root/leaf clusters seem to be significant enough to be encoded in a more general API instead of relying on ad-hoc name comparison.

2. I realize that it's probably a given but it wasn't entirely clear that cluster names are unique. Mere cluster name comparison could therefore possibly yield multiple clusters marked as "root". My alternative seems to prevent this possibility quite clearly.

3. Your code snippet also demonstrates why we should prefer to encode as much information as possible in types and code structure: the old `GetAllClusters` seemed to intend to imply an invariant that the root cluster *must* be listed as the first entry. However, since this is not exposed/guaranteed in any way, your code performs this sorting "defensively" once again.

I originally contemplated an option to retain `GetAllClusters` but change its return type into `[](struct {services.RemoteCluster; isRoot bool})` or some such. But then, I think it's a bad practice to rely on boolean flags too much - their meaning is often difficult to trace and they tend to be difficult to extend to multiple choices/values.

**Comment by andrejtokarcik** [comment]
@awly Note that `GetRootCluster` doesn't wrap only the cluster's name but also the connection status (as well as the last heartbeat but that doesn't seem to come into play at the moment). Once again, I think it's better to provide this info uniformly to any consumer, even though it might be just the `tsh clusters` handler at this point.

To avoid the impression that we're fighting the type system we could perhaps just abstract away those methods of `RemoteCluster` that are relevant for the root cluster into a different struct/interface `Cluster` and to embed this into `RemoteCluster`.

**Comment by awly** [comment]
The connection status and last heartbeat are set to constant values and we're not likely to change that (if you're talking to the root cluster, it's obviously online and last seen at the time of request).
Those are only set to shoehorn the root cluster into the `services.RemtoeCluster` type.

File: tool/tsh/tsh.go (hunk 2 above)
Comment (webvictim) [reply tool/tsh/tsh.go]: We don't make this very obvious in our docs, so better suggestions may be welcomed :)

Comment (andrejtokarcik) [reply tool/tsh/tsh.go]: Kept the column but renamed it to 'Selected' as recommended by @benarent in #4965.

**Review by awly — APPROVED** [review]

**Pull request merged**